Material for MkDocs

jodygarnett opened the pull request:
Content converted using pandoc, this is a prototype ahead of boldens code sprint.

I am neutral on MkDocs, but appreciate markdown as it is used in so many other places. Material for MkDocs has so many extensions that we can get 90% of sphinx-build funcitonality. 

The only real weakness is including live examples which is done using line-number range (making it hard to fix documentation if a configuration file being documented changes).

```
pip3 install -r requirements.txt
mkdocs serve
open http://localhost:8000
```

See README.md in this pull request for alternate setup instructions, or consult referencess below:

* https://squidfunk.github.io/mkdocs-material/getting-started/
* https://squidfunk.github.io/mkdocs-material/reference/
* https://www.mkdocs.org/user-guide/writing-your-docs/

<img width="1221" alt="image" src="https://github.com/metadata101/iso19139.ca.HNAP/assets/629681/b310342b-27c2-48f0-9e87-fa3c39030ffc">

Compare:

* https://jodygarnett.github.io/iso19139.ca.HNAP/
* https://metadata101.github.io/iso19139.ca.HNAP/

jodygarnett committed:
remove docs folder, moving to gh-pages branch

jodygarnett committed:
initial conversion to markdown using pandoc

Manual clean with search and repalce to address warnings

jodygarnett committed:
cleanup markdown formatting

jodygarnett commented:
I think I have a live preview now at https://jodygarnett.github.io/iso19139.ca.HNAP/

jodygarnett committed:
navigation improvements

jodygarnett committed:
Switch to bold for user interface headings, fields, buttons

ianwallen commented on .gitignore:
```diff
@@ -27,6 +27,11 @@ target/
+venv
+site
+
 # Sphinx-build
```
If you are getting ride of Sphinx do we still need this?

ianwallen commented on pom.xml:
```diff
@@ -216,6 +216,7 @@
               </execution>
             </executions>
           </plugin>
+          <!--
```
If not required then shouldn't it be removed?

jodygarnett commented:
I will remove sphinx-build when I have mkdocs working. I need to sort out translation … and it is being very difficult.

jodygarnett committed:
Update content prior to translation

jodygarnett committed:
Update screen snaps for search documentation

jodygarnett committed:
Update screen snaps for edit documentation

jodygarnett committed:
Outline mapping documentation

jodygarnett committed:
remove sphinx-build documentaton and translations

jodygarnett committed:
workflow automation deploy to gh-pages

jodygarnett committed:
update build to use mkdocs rather than sphinx-build

This removes the -Pdocs proifile, ant checks for mkdocs availability and will provide a warning if it is not avaiable.

# Conflicts:
#	pom.xml

jodygarnett committed:
First attempt at automation for deepl api translation of individual files

jodygarnett committed:
Add development section to manual to cover documentation writing guide and translation

jodygarnett committed:
Adjust pre/post processing in translate script to allow roundtrip from markdown to html

jodygarnett committed:
Initial french translation

jodygarnett committed:
Use CDATA to force deepl to preserve code block whitespce formatting

jodygarnett committed:
translate french as one command

jodygarnett committed:
French translation test

jodygarnett committed:
translate search/details to french

jodygarnett commented:
This content has moved to core-geonetwork
